Cake.Recipe/Content/build.cake

```diff
@@ -170,44 +170,32 @@ BuildParameters.Tasks.BuildTask = Task("Build")
     .Does<BuildData>(data => RequireTool(MSBuildExtensionPackTool, () => {
         Information("Building {0}", BuildParameters.SolutionFilePath);
 
-        if(BuildParameters.IsRunningOnWindows)
-        {
-            var msbuildSettings = new MSBuildSettings()
-                .SetPlatformTarget(ToolSettings.BuildPlatformTarget)
-                .UseToolVersion(ToolSettings.BuildMSBuildToolVersion)
-                .WithProperty("TreatWarningsAsErrors", BuildParameters.TreatWarningsAsErrors.ToString())
-                .WithTarget("Build")
-                .SetMaxCpuCount(ToolSettings.MaxCpuCount)
-                .SetConfiguration(BuildParameters.Configuration)
-                .WithLogger(
-                    Context.Tools.Resolve("MSBuild.ExtensionPack.Loggers.dll").FullPath,
-                    "XmlFileLogger",
-                    string.Format(
-                        "logfile=\"{0}\";invalidCharReplacement=_;verbosity=Detailed;encoding=UTF-8",
-                        BuildParameters.Paths.Files.BuildLogFilePath)
-                );
-
-            MSBuild(BuildParameters.SolutionFilePath, msbuildSettings);
-
-            // Parse warnings.
-            var issues = ReadIssues(
-                MsBuildIssuesFromFilePath(
-                    BuildParameters.Paths.Files.BuildLogFilePath,
-                    MsBuildXmlFileLoggerFormat),
-                data.RepositoryRoot);
-
-            Information("{0} MsBuild warnings are found.", issues.Count());
-            data.AddIssues(issues);
-        }
-        else
-        {
-            var xbuildSettings = new XBuildSettings()
-                .SetConfiguration(BuildParameters.Configuration)
-                .WithTarget("Build")
-                .WithProperty("TreatWarningsAsErrors", "true");
-
-            XBuild(BuildParameters.SolutionFilePath, xbuildSettings);
-        }
+        var msbuildSettings = new MSBuildSettings()
+            .SetPlatformTarget(ToolSettings.BuildPlatformTarget)
+            .UseToolVersion(ToolSettings.BuildMSBuildToolVersion)
+            .WithProperty("TreatWarningsAsErrors", BuildParameters.TreatWarningsAsErrors.ToString())
+            .WithTarget("Build")
+            .SetMaxCpuCount(ToolSettings.MaxCpuCount)
+            .SetConfiguration(BuildParameters.Configuration)
+            .WithLogger(
+                Context.Tools.Resolve("MSBuild.ExtensionPack.Loggers.dll").FullPath,
+                "XmlFileLogger",
+                string.Format(
+                    "logfile=\"{0}\";invalidCharReplacement=_;verbosity=Detailed;encoding=UTF-8",
+                    BuildParameters.Paths.Files.BuildLogFilePath)
+            );
+
+        MSBuild(BuildParameters.SolutionFilePath, msbuildSettings);
```

Comment:
I actually disagree with just running msbuild blindly.
I would say it would be appropriate to detect if available first (on non-windows builds), and if it is then use msbuild, otherwise use xbuild.

Not all linux distros have msbuild as a package in the official repository.
For instance, on Arch Linux there is a AUR package for msbuild, but you need to compile MSBuild with that package, and can not rely on pre-built packages.

There are probably other distros that don't have it as well, but I only know what is available on Arch.

Reply:
I don't agree on running xbuild blindly either, since it is broken for anything using SDK style projects.

But you are right. It seems like several distros don't install msbuild by default.

Reply:
> since it is broken for anything using SDK style projects.

That is not entirely true, I have ran xbuild successfully on SDK style projects, however there are quite a few warnings occurring when doing that though.

Personally I have usually used the dotnet core utility though when running SDK styled projects, even if the project itself don't target .NET Core.

> I don't agree on running xbuild blindly either

True, running it blindly isn't the best, however using it as a fallback if msbuild isn't available would be appropriate IMO. There really isn't much else to fallback to after msbuild and xbuild have been checked, so it shouldn't make much different by falling back to xbuild.